harness/determined/pytorch/_reducer.py

```diff
@@ -184,7 +185,17 @@ def per_slot_reduce(self) -> Any:
         return self.values
 
     def cross_slot_reduce(self, per_slot_metrics: List) -> Any:
-        flat_metrics = [item for sublist in per_slot_metrics for item in sublist]
+        """
+        Call the reducer function on metrics in their original order.
+
+        Interleave metrics from different slots as we flatten
+        the list of metrics to undo the effect of sharding.
+        Note: this will only reconstruct the original order if the user
+        calls ``update()`` the same number of times per batch
+        """
+        interleaved = itertools.zip_longest(*per_slot_metrics)
+        flat_metrics = [m for sublist in interleaved for m in sublist if m is not None]
```

Comment:
Oh, I think we're silently preventing users from passing a None value.  Can we use some sort of sentinel that is not None?

```python
class NonMetric:
   pass

nonmetric = NonMetric()

interleaved = itertools.zip_longest(*per_slot_metrics, fillvalue=not_metric)
flat_metrics = [m for sublist in interleaved for m in sublist if m is not nonmetric]
```

Reply:
wow, good catch.

Reply:
lmk if the scoping I chose for `NotMetric` is ok.